WordPress/Classes/ViewRelated/Reader/ReaderSiteStreamHeader.swift

```diff
@@ -29,25 +34,89 @@ import Foundation
+        // Vertical and horizontal margins
+        let hMargin = descriptionLabel.frame.minX
+        let vMargin = descriptionLabel.frame.minY
+
+        let width = UIDevice.isPad() ? contentIPadWidthConstraint!.constant : size.width
+        let innerWidth = width - (hMargin * 2)
+        let adjustedSize = CGSize(width:innerWidth, height:CGFloat.max)
+
+        var height = UIDevice.isPad() ? contentIPadTopConstraint!.constant : 0
+        height += descriptionView.frame.minY
+        height += vMargin
+        height += descriptionLabel.sizeThatFits(adjustedSize).height
+        height += descriptionBottomConstraint.constant
+        height += followCountLabel.sizeThatFits(adjustedSize).height
+        height += followCountBottomConstraint.constant;
+
+        return CGSize(width: size.width, height: height)
+    }
+
 
    // MARK: - Configuration
 
     public func configureHeader(topic: ReaderAbstractTopic) {
-        // TODO: Wire up actual display when supported in core data
-        avatarImageView.setImageWithURL(NSURL(), placeholderImage: UIImage(named: defaultBlavatar))
-        titleLabel.text = topic.title
-        detailLabel.text = "site.com"
-        if topic.following {
+        assert(topic.isKindOfClass(ReaderSiteTopic), "Topic must be a site topic")
+
+        let siteTopic = topic as! ReaderSiteTopic
+
+        configureHeaderImage(siteTopic.siteBlavatar)
+
+        titleLabel.text = siteTopic.title
+        detailLabel.text = NSURL(string: siteTopic.siteURL)?.host
+        if siteTopic.following {
             WPStyleGuide.applyReaderStreamHeaderFollowingStyle(followButton)
         } else {
             WPStyleGuide.applyReaderStreamHeaderNotFollowingStyle(followButton)
         }
 
-        followCountLabel.text = "100 followers"
-        descriptionLabel.text = "Just another WordPress site"
-//        var attributes = WPStyleGuide.readerStreamHeaderDescriptionAttributes()
-//        var attributedText = NSAttributedString(string: topic.description, attributes: attributes)
-//        descriptionLabel.attributedText = attributedText
+        descriptionLabel.attributedText = attributedSiteDescriptionForTopic(siteTopic)
+
+        if descriptionLabel.attributedText?.length > 0 {
+            // Bottom and top margins should match.
+            descriptionBottomConstraint.constant = descriptionLabel.frame.minY
+        } else {
+            descriptionBottomConstraint.constant = 0
+        }
+
+
+        followCountLabel.text = formattedFollowerCountForTopic(siteTopic)
+    }
+
+    func configureHeaderImage(siteBlavatar: String?) {
+        let placeholder = UIImage(named: defaultBlavatar)
+
+        var path = ""
+        if siteBlavatar != nil {
+            path = siteBlavatar!
+        }
+
+        let url = NSURL(string: path)
+        if url != nil {
+            avatarImageView.setImageWithURL(url!, placeholderImage: placeholder)
+        } else {
+            avatarImageView.image = placeholder
+        }
+    }
```

Comment:
Code looks correct, but I just wanted to share that I'm lately enjoying map/flatMap for optional handling. Like this:

```swift
    func configureHeaderImage(siteBlavatar: String?) {
        let placeholder = UIImage(named: defaultBlavatar)

        if let url: NSURL = siteBlavatar.flatMap({ NSURL(string: $0) }) {
            avatarImageView.setImageWithURL(url, placeholderImage: placeholder)
        } else {
            avatarImageView.image = placeholder
        }
    }
```